tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -99,6 +99,7 @@ internal TracerManager CreateTracerManager(ImmutableTracerSettings settings, Tra
             settings ??= ImmutableTracerSettings.FromDefaultSources();
 
             var defaultServiceName = settings.ServiceName ??
+                ServerlessMiniAgent.GetGCPAzureFunctionName() ??
```

Comment:
🤔 Should we change the service name? Wondering because I would have expected to have the same behaviour as in Azure Functions and we don't do that there. So maybe we were wrong obviously, but still feels weird. @lucaspimentel is way better than me on those things so I'll let him agree/disagree with me.

Reply:
Also, in case you don't need it here, you could rely only on ImmutableTracerSettings because all other use cases happen after the settings are built

Reply:
You're right, re-visiting this i think it makes the most sense to leave application + service name untouched. Reverted